### PR TITLE
misc-gitology: init at unstable-2023-06-22

### PR DESCRIPTION
**Pull request by 9999years**
## Description of changes

A collection of useful Git scripts, notably including `git-flip-history` (which lets you split a commit by writing a series of commits _undoing_ the changes one by one).

See: https://blog.aloni.org/posts/gitology-1-git-flip-history/

Upstream repo: https://github.com/da-x/misc-gitology

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 9999years** [comment]
Note that I haven't included `git` or any of the runtime dependencies in this package. My understanding is that the scripts are meant to use the version of `git` that a user has installed. Does that sound OK, or should I add `git` to the runtime dependencies here?

File: pkgs/applications/version-management/misc-gitology/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    mkdir -p $out/bin
+    find . \
+      -type f \
+      -executable \
+      -maxdepth 1 \
+      -exec install --target-directory=$out/bin/ {} +
+  '';
+
+  meta = with lib; {
+    description = " An assortment of scripts around Git";
```
Comment (FliegendeWurst) [inline pkgs/applications/version-management/misc-gitology/default.nix:29]:
```suggestion
    description = "Assortment of scripts around Git";
```

File: pkgs/applications/version-management/misc-gitology/default.nix
Comment (FliegendeWurst) [inline pkgs/applications/version-management/misc-gitology/default.nix:1]: Should be in pkgs/by-name.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/mi/misc-gitology/package.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  nix-update-script,
+  perl,
+  python3,
+}:
+stdenv.mkDerivation {
+  pname = "misc-gitology";
+  version = "unstable-2024-08-26";
```
Comment (FliegendeWurst) [inline pkgs/by-name/mi/misc-gitology/package.nix:11]:
```suggestion
  version = "0-unstable-2024-08-26";
```
(new policy to always start version with a digit)

File: pkgs/by-name/mi/misc-gitology/package.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  dontBuild = true;
+
+  buildInputs = [
+    python3
+    # For `git-find-blob`:
+    perl
+  ];
+
+  installPhase = ''
+    mkdir -p $out/bin
```
Comment (FliegendeWurst) [inline pkgs/by-name/mi/misc-gitology/package.nix:29]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/bin
```

File: pkgs/by-name/mi/misc-gitology/package.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    find . \
+      -type f \
+      -executable \
+      -maxdepth 1 \
+      -exec install --target-directory=$out/bin/ {} +
+  '';
```
Comment (FliegendeWurst) [inline pkgs/by-name/mi/misc-gitology/package.nix:35]:
```suggestion
    runHook postInstall
  '';
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**